### ISSUE TITLE
Add venv into docker image to enable editable `pip install`

### ISSUE THREAD

**Issue by siriuslee** [mosaicml/composer#180]
When trying to install composer with `pip install -e .` from within the docker image, we are seeing this error:
```
Traceback (most recent call last):
  File "/usr/bin/composer", line 33, in <module>
    sys.exit(load_entry_point('mosaicml', 'console_scripts', 'composer')())
  File "/usr/bin/composer", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/usr/lib/python3.8/importlib/metadata.py", line 445, in distribution
    return Distribution.from_name(distribution_name)
  File "/usr/lib/python3.8/importlib/metadata.py", line 169, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: mosaicml
```
This seems to be remedied by running the `pip install` from within a virtualenv. Can we bake a virtualenv into the docker image as a workaround?

**Comment by Averylamp** [comment]
I found this issue recently which actually was messing up my local installs which I thought was interesting

https://github.com/pypa/pip/issues/7953

**Comment by ravi-mosaicml** [comment]
This is fixed in #182 -- we needed to update setuptools and modify the setup.py
